backend/index.js

```diff
@@ -0,0 +1,53 @@
+const express = require('express');
+const { exec } = require('child_process');
+const app = express();
+const cors = require('cors')
+
+function runShellCommand(command, callback) {
+    exec(command, (error, stdout, stderr) => {
+        if (error) {
+            callback(`Error: ${error.message}`);
+            return;
+        }
+        if (stderr) {
+            callback(stderr);
+            return;
+        }
+        callback(stdout);
+    });
+}
+
+function extractStack(output) {
+    const match = output.match(/\[.*\]/);
+    return match ? match[0] : 'No message found';
+}
+
+app.use(cors());
+app.use(express.json()); 
+
+app.post('/run-script', (req, res) => {
+    const { pub_key, sig } = req.body;
+    if (!pub_key) {
+        return res.status(400).send('Missing public key parameter');
+    }
+    const scriptPath = '../tests/text_to_byte_array.sh';
```

Comment:
Let's add some input validation to ensure the pub_key and sig aren't massive strings we ask the backend to process. Maybe limiting the length of each.

The max script size is 10000 bytes and the longest named opcode is about 25 characters. So I think we can say the pub_key and sig should be less than 350000 ( 25 * 10000 = 250000 + a small extra allowance )

Please leave a comment on how the max size was defined to for a reminder

Reply:
Okay boss